### PR TITLE
Add support for multiple submit plugins

### PR DESCRIPTION
**Pull request by pschorf**
## Changes proposed in this PR
- Adds support for multiple submit plugins

## Why are we making these changes?
To use multiple submit plugins

### CODE REVIEW

File: scheduler/src/cook/plugins/submission.clj [hunk 1]
```diff
@@ -38,21 +37,57 @@
...
+                                             resolved
+                                             (throw (ex-info "Unable to resolve factory function" factory-fn))))
+                                   factory-fns))
+          plugins (map (fn [f] (f)) resolved-fns)]
+      (log/info "Creating composite plugin with components" resolved-fns)
+      (->CompositeSubmissionPlugin plugins))
+    (if-let [resolved-fn (cook.plugins.util/resolve-symbol (symbol factory-fn))]
+      (do
+         (log/info "Using plugin" resolved-fn)
+        (resolved-fn))
```
Comment (dposada) [inline scheduler/src/cook/plugins/submission.clj]: Indentation seems off on this line

File: integration/tests/cook/util.py [hunk 2]
```diff
@@ -1305,7 +1305,7 @@ def demo_plugin_is_configured(cook_url):
     settings_dict = settings(cook_url)
     # Because we always create plugin configuration in config.clj, the first keys always exist.
     # The actual factory-fn keys are not set unless the user specifies them.
-    if settings_dict['plugins']['job-submission-validator'].get('factory-fn') != "cook.plugins.demo-plugin/submission-factory":
+    if settings_dict['plugins']['job-submission-validator'].get('factory-fn') == ["cook.plugins.demo-plugin/submission-factory", "cook.plugins.demo-plugin/submission-factory2"]:
```
[the remark this answers is not in the document]
Comment (pschorf) [reply integration/tests/cook/util.py]: Nice catch, fixed

File: integration/tests/cook/util.py [hunk 3]
```diff
@@ -1305,7 +1305,7 @@ def demo_plugin_is_configured(cook_url):
...
         return False
```
[the remark this answers is not in the document]
Comment (pschorf) [reply integration/tests/cook/util.py]: I think the bug was actually with the `==`

File: integration/tests/cook/util.py [hunk 4]
```diff
@@ -1305,7 +1305,7 @@ def demo_plugin_is_configured(cook_url):
     settings_dict = settings(cook_url)
```
[the remark this answers is not in the document]
Comment (pschorf) [reply integration/tests/cook/util.py]: Done

File: scheduler/src/cook/plugins/submission.clj (hunk 1 above)
Comment (pschorf) [reply scheduler/src/cook/plugins/submission.clj]: Done

**Pull request merged**